languageServer/src/textDocument.ts

```diff
@@ -17,19 +17,11 @@ import {
   visit
 } from 'yaml'
 
-export const getTextDocumentLocation = (
-  textDocument: TextDocument
-): Location => {
-  const start = Position.create(0, 0)
-  const end = textDocument.positionAt(textDocument.getText().length - 1)
-  const range = Range.create(start, end)
-
-  return Location.create(textDocument.uri, range)
-}
+export const getLocationToOpen = (uri: string): Location => {
+  const position = Position.create(0, 0)
+  const range = Range.create(position, position)
```

Comment:
[F] Turns out that we can open the file using an empty range (previously this didn't work or I had done the wrong thing to test). This means we don't need the file contents at all (for now). We can simply check whether or not the URI relates to a file. If it does then we can create a location with the URI string and an empty range. Passing this back to the client will open the file. If there are multiple locations (files) then the user will be prompted which one they want to open. That's the reason that we can't just ask the extension to open the file without first returning information to the server.

Reply:
We also have to check whether or not the file is a file on the extension side because the LSP might not be running in a context that can access the file system (e.g Codespaces)